components/org.wso2.carbon.identity.sts.passive.ui/src/main/java/org/wso2/carbon/identity/sts/passive/ui/PassiveSTS.java

```diff
@@ -345,9 +347,21 @@ private void persistWreply(ResponseToken responseToken, HttpSession session) {
     private void sendToAuthenticationFramework(HttpServletRequest request, HttpServletResponse response,
                                                String sessionDataKey, SessionDTO sessionDTO) throws IOException {
 
-        String commonAuthURL = IdentityUtil.getServerURL(FrameworkConstants.COMMONAUTH, false, true);
+        String commonAuthURL;
+        try {
+            commonAuthURL = ServiceURLBuilder.create().addPath(FrameworkConstants.COMMONAUTH).build()
+                    .getAbsolutePublicURL();
+        } catch (URLBuilderException e) {
+            throw new IOException("Error occurred while building the commonauth URL during login.", e);
```

Comment:
don't we have a specific exception type added in passive sts component.
As this is a private method shall we specify and exception type and utilize that rather than using IOException.

Reply:
addressed